### ISSUE TITLE
Regarding angular2-query-builder, I want the field dropdown to be customizable

### ISSUE THREAD

**Issue by kritisharmaa01** [zebzhao/Angular-QueryBuilder#259]
While using angular2-query-builder, I can change the type of OPTIONS of a FIELD. I can even add my customized type for OPTIONS. But, for the FIELDS I can't make any such changes. Kindly guide me on that. I have a requirement where both the dropdowns should be such that: I should be allowed to type text(alphanumeric). Such that I can search in the dropdown. I have a custom type for this which works similar to DATALIST.

But I can't understand how to embed my change in angular2-query-builder such that I can make changes to FIELD DROPDOWN.

It is an urgent requirement. I request you to please, respond asap if possible

**Comment by fourthofaugust** [comment]
@kritisharmaa01 I have a similar requirement. Have you tried using the `queryField` directive to customize the field selection?

**Comment by kritisharmaa01** [comment]
@fourthofaugust  No, but I tried working on 'parentFieldTemplate' attribute. It had no impact on the template. Did the queryField directive customization work for you?
linkedin.com/in/kriti-sharma18/

**Comment by fourthofaugust** [comment]
I was playing around and it looks that directive would work. There's a demo app from a different fork called ngx-angular-query-builder. May be you can give it a shot there.
